src/Spain/Mews.Fiscalizations.Spain/Model/Request/ForeignCompany.cs

```diff
@@ -1,16 +1,19 @@
-﻿namespace Mews.Fiscalizations.Spain.Model.Request
+﻿using Mews.Fiscalizations.Core.Model;
+
+namespace Mews.Fiscalizations.Spain.Model.Request
 {
     public sealed class ForeignCompany
     {
-        public ForeignCompany(Name name, ForeignCompanyTaxpayerNumber taxpayerNumber)
+        public ForeignCompany(Name name, ForeignTaxpayerNumber taxpayerNumber)
         {
-            Name = name;
-            TaxpayerNumber = taxpayerNumber;
+            Name = Check.IsNotNull(name, nameof(name));
+            TaxpayerNumber = Check.IsNotNull(taxpayerNumber, nameof(taxpayerNumber));
+            Check.Condition(!TaxpayerNumber.Country.Equals(Countries.Spain), "Foreign company cannot be Spanish.");
```

Comment:
This check should be in the Create method of the ForeignTaxPayerNumber.

It's called foreign. So if we can create it with non-foreign, that's the wrong part.